### PR TITLE
Issue 17785 - fix search with sort for reactive and elasticsearch options

### LINKED ISSUE #17785: Elasticsearch exception No mapping found for [date.keyword] in order to sort on
##### **Overview of the issue**

Search for entities that had paginate in the JDL, after sorting by date, fail with the following exception:

```
Suppressed: org.elasticsearch.ElasticsearchException: Elasticsearch exception [type=query_shard_exception, reason=No mapping found for [date.keyword] in order to sort on]
```

##### **Motivation for or Use Case**

In entities that had `paginate`in the JDL, after sorting the table by a non-text field, subsequent search fails, as the query builder assumes the sort field has multi-field mapping, and has a keyword field for sorting. Keyword field is only created automatically for text fields.

##### **Reproduce the error**

```
jhipster jdl blog-reactive-ms.jdl
cd gateway
./mvnw -ntp -Pprod verify jib:dockerBuild
cd ../blog
./mvnw -ntp -Pprod verify jib:dockerBuild
cd ../docker-compose
docker compose up
```

1. Go to localhost:8080, sign in
2. Go to entities -> Blog
3. Create a Blog
4. Go to entities -> Post
5. Create a Post
6. Sort the table by Date
7. Search for a Post -> Internal Server Error

##### **Related issues**

Related to #17018

##### **JHipster Version(s)**

```
INFO! Using JHipster version installed globally
7.6.0
```

##### **JHipster configuration**

  <details>
  <summary>JDL definitions</summary>
  <pre>
application {
  config {
    baseName gateway
    reactive true
    packageName com.okta.developer.gateway
    applicationType gateway
    authenticationType oauth2
    buildTool maven
    clientFramework vue
    prodDatabaseType postgresql
    serviceDiscoveryType eureka
    testFrameworks [cypress]
  }
  entities Blog, Post, Tag
}
application {
  config {
    baseName blog
    reactive true
    packageName com.okta.developer.blog
    applicationType microservice
    authenticationType oauth2
    buildTool maven
    databaseType mongodb
    devDatabaseType mongodb
    prodDatabaseType mongodb
    enableHibernateCache false
    serverPort 8081
    serviceDiscoveryType eureka
    searchEngine elasticsearch
  }
  entities Blog, Post, Tag
  service * with serviceClass
  search * with elasticsearch
}
entity Blog {
  name String required minlength(3)
  handle String required minlength(2)
}
entity Post {
  title String required
  content TextBlob required
  date Instant required
}
entity Tag {
  name String required minlength(2)
}
relationship ManyToOne {
  Blog{user(login)} to User
  Post{blog(name)} to Blog
}
relationship ManyToMany {
  Post{tag(name)} to Tag{post}
}
paginate Post, Tag with pagination
microservice Blog, Post, Tag with blog
deployment {
  deploymentType docker-compose
  appsFolders [gateway, blog]
  dockerRepositoryName "jgarbarino"
}

  </pre>
  </details>

##### **Browsers and Operating System**

Ubuntu 18.04.6 LTS

- [ x ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by maznag**
Proposed PR to fix #17785.

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/entity-server/files.js [hunk 1]
```diff
@@ -172,16 +172,6 @@ const serverFiles = {
         },
       ],
     },
-    {
-      condition: generator => generator.searchEngine === ELASTICSEARCH && !generator.embedded && !generator.paginationNo,
-      path: SERVER_MAIN_SRC_DIR,
-      templates: [
-        {
-          file: 'package/repository/search/SortToSortBuilderListConverter.java',
```
Comment (DanielFran) [inline generators/entity-server/files.js:180]: Can you cleanup the file here: https://github.com/jhipster/generator-jhipster/blob/main/generators/server/index.js#L457

**Review by DanielFran — CHANGES_REQUESTED** [review]

File: generators/entity-server/files.js (hunk 1 above)
Comment (DanielFran) [reply generators/entity-server/files.js:180]: @maznag You will need to rebase also

Comment (maznag) [reply generators/entity-server/files.js:180]: done !

**Comment by DanielFran** [comment]
@maznag something wrong. Can you rebase?

**Comment by maznag** [comment]
> @maznag something wrong. Can you rebase?

Ooops ! sorry, I completely create new branch from my last commit before previous rebase and do new rebase. I don't know what happened on my last rebase :-). I think everything is ok now

**Comment by DanielFran** [comment]
@maznag By mistake, I send you the index.js path to server generator instead of entity-server generator.
I fixed the removal of SortToSortBuilderListConverter class directly in your branch

**Review by DanielFran — APPROVED** [review]

**Comment by DanielFran** [comment]
Thnks @maznag for the contribution

**Comment by maznag** [comment]
Thank you to you too @DanielFran :-)

**Comment by maznag** [comment]
@pascalgrimaud there is a bounty on this ticket. This is the link to OpenCollective expense https://opencollective.com/generator-jhipster/expenses/68435. Thanks in advance !

**Comment by pascalgrimaud** [comment]
@maznag : approved, thanks for your work ;)

**Pull request merged**